src/librustc/middle/traits/error_reporting.rs

```diff
@@ -230,6 +230,26 @@ pub fn report_selection_error<'a, 'tcx>(infcx: &InferCtxt<'a, 'tcx>,
                                 "the trait `{}` is not implemented for the type `{}`",
                                 trait_ref, trait_ref.self_ty());
 
+                            let mut counter = 1;
+                            infcx.tcx.sess.fileline_help(
+                                obligation.cause.span,
+                                "the following implementations were found:");
```

Comment:
The way I'm reading this, it looks like "the following implementations were found:" will be printed, even if there are no other applicable implementations. I think output like:

```
help: the following implementations were found:
```

would be quite confusing.